### PR TITLE
MM-29636 - add the cancel subscription section in subscription screen

### PR DESCRIPTION
**Pull request by pablovelezvidal**
#### Summary
This Merge Request adds the cancel subscription section to the subscription page under the System Console > Billing & Account > Subscription

#### Ticket Link
This MR fixes [MM-29636](https://mattermost.atlassian.net/browse/MM-29636)

#### Screenshots
Captura de pantalla 2021-01-08 a las 13 52 42
Captura de pantalla 2021-01-08 a las 13 52 32

### CODE REVIEW

File: components/admin_console/billing/billing_subscriptions.scss [hunk 1]
```diff
@@ -196,3 +196,57 @@
         text-decoration: none;
     }
 }
+
+.customerCancelSubscriptionSection {
```
Comment (marianunez) [inline components/admin_console/billing/billing_subscriptions.scss:200]: nit: I would drop the word `customer` on all these names so they are not so wordy

```suggestion
.cancelSubscriptionSection {
```

**Review by marianunez — APPROVED** [review]
Looks great! Thanks @pablovelezvidal!

**Comment by stevemudie** [comment]
@pablovelezvidal when got to [Billing & Account ➜ Subscription](https://mattermost-webapp-pr-7295.test.mattermost.cloud/admin_console/billing/subscription) on the spinwick it's blank. Is that expected due to the environment here?

**Comment by marianunez** [comment]
> @pablovelezvidal when got to [Billing & Account ➜ Subscription](https://mattermost-webapp-pr-7295.test.mattermost.cloud/admin_console/billing/subscription) on the spinwick it's blank. Is that expected due to the environment here?

Yes, I am setting up the PR server manually so you can access that. Will keep you posted when is ready

File: components/admin_console/billing/billing_subscriptions.scss (hunk 1 above)
Comment (pablovelezvidal) [reply components/admin_console/billing/billing_subscriptions.scss]: agree. Done!

**Comment by marianunez** [comment]
> @pablovelezvidal when got to [Billing & Account ➜ Subscription](https://mattermost-webapp-pr-7295.test.mattermost.cloud/admin_console/billing/subscription) on the spinwick it's blank. Is that expected due to the environment here?

@stevemudie @matthewbirtch I've updated the test server here and now you should be able to see the Subscription section in the System Console to review the change.

**Comment by stevemudie** [comment]
Looks good but this will only appear when there's a subscription in place though right?

Also, would it make sense to have the contact form pre-fill the 'What can sales help with?' section with the relevant 'I want to cancel my Mattermost account' selection as it already pre-selects 'I need to contact sales' now?

On that note, since the form lacks a "support" dropdown (there's only technical billing & sales) should be reworked to include it? I'm just thinking it might be a point of confusion since the customer clicked the "Contact Support" button and the form sets them up with "I need to contact **sales**" automatically.

**Review by matthewbirtch — APPROVED** [review]
Looks great to me! nice work @pablovelezvidal. 

Agree with @stevemudie about the possibility of updating the contact form to prefill with 'I want to cancel my subscription' or something along those lines, but that's outside of the scope of this PR.

**Comment by matthewbirtch** [comment]
One question I had in the screenshot provided in the description, how did you get a state where there were no subscription details to the left of the 'Need more users' callout?
image

**Comment by marianunez** [comment]
> I'm just thinking it might be a point of confusion since the customer clicked the "Contact Support" button and the form sets them up with "I need to contact sales" automatically.

Agreed, @matthewbirtch thoughts on wording? 

At the same time it's odd to call it `Contact Sales` when you want to cancel. Maybe we can move the option to cancel your subscription to the technical request in the cloud contact us page instead of under `I have a sales request`?

Spec for reference: https://docs.google.com/document/d/1O_2d5cWtH3-8sq2wdHuHpfbSbxjPkfuzNJTdas71dRo/edit#bookmark=id.c7xmi1v4ly1h

**Comment by matthewbirtch** [comment]
@marianunez shouldn't it preselect the "I want to cancel my Mattermost account" option though? It doesn't seem to be doing that for me.
image

As for the language confusion, one simple way to alleviate some of the confusion would be to simply change the button from 'Contact Support' to 'Contact Us' on system console cancelation section. The resulting page title is 'Contact Us' anyway

**Comment by marianunez** [comment]
> shouldn't it preselect the "I want to cancel my Mattermost account" option though? It doesn't seem to be doing that for me.

Yes, it should but currently CWS doesn't support the second parameter to be preselected. I will create a follow-up ticket to implement outside of this PR. 

> As for the language confusion, one simple way to alleviate some of the confusion would be to simply change the button from 'Contact Support' to 'Contact Us' on system console cancelation section. The resulting page title is 'Contact Us' anyway

+1 on 'Contact Us' wording. @pablovelezvidal could you update the button text to that?

**Review by nickmisasi — APPROVED** [review]
LGTM! Great work @pablovelezvidal!

**Review by stevemudie — APPROVED** [review]
Looks good!

**Comment by pablovelezvidal** [comment]
> @marianunez shouldn't it preselect the "I want to cancel my Mattermost account" option though? It doesn't seem to be doing that for me.
> image
> 
> As for the language confusion, one simple way to alleviate some of the confusion would be to simply change the button from 'Contact Support' to 'Contact Us' on system console cancelation section. The resulting page title is 'Contact Us' anyway

The ticket to follow up this PR is [MM-31935](https://mattermost.atlassian.net/browse/MM-31935)

**Pull request merged**